[POC] Version 2.0 (#21)
terminal42/contao-inserttags: README.md

# terminal42/contao-inserttags

> Adds a backend module to define your custom insert tags. You can limit an insert tag to certain pages,

Reviewer: ```suggestion
Adds a back end module to define your custom insert tags. You can limit an insert tag to certain pages,
```

> show it to guests, or only for the logged in frontend users.

Reviewer: ```suggestion
show it to guests only, or restrict it to certain logged in front end member groups.
```

## Upgrade from version 1.x

Some options present in version 1.x have been removed. All changes are described below.  

### Removed "timing"

> The timing feature has been removed completely, because it would kill the HTTP caching. If you need a replacement,

Reviewer: ```suggestion
The timing feature has been removed completely, because it would kill HTTP caching. If you need a replacement,
```

> please consider the JavaScript solution with an optional AJAX request(s).

Reviewer: ```suggestion
please consider a JavaScript solution.
```

### Removed "limitLanguages"

This option has been removed completely, use the limiting insert tags to certain pages feature instead.

### Removed "useCondition"

> This option has been removed completely.

Reviewer: ```suggestion
This option has been removed completely as this too, would make HTTP caching impossible.
```

### Removed "useCounter"

> This option has been removed completely.

Reviewer: ```suggestion
This option has been removed completely as this too, would make HTTP caching impossible.
```

### Removed "mode"

> This option has been removed completely, as insert tags are only present in the frontend.

Reviewer: ```suggestion
This option has been removed completely, as insert tags are only present in the front end.
```

### Removed "cacheOutput"

> This option has been removed completely.

Reviewer: ```suggestion
This option has been removed completely as it is not required in version 2 anymore.
```

# terminal42/contao-inserttags

Adds a back end module to define your custom insert tags. You can limit an insert tag to certain pages,
show it to guests only, or restrict it to certain logged in front end member groups.

The extension supports the simple tokens parsing and the [Expression Language](https://symfony.com/doc/current/components/expression_language.html#how-can-the-expression-engine-help-me) component,
which allows to create powerful and dynamic replacements.

```
{if 123 in page.trail}
I am a subpage of page ID 123!
{endif}

{if 456 in user.groups}
I belong to member group ID 456!
{endif}

{if request.query.has('foobar')}
My "foobar" parameter is: ##request.query.get('foobar')##
{else}
I have no "foobar" parameter :(
{endif}
```

## Supported simple tokens

Below you can find a list of supported simple tokens:

### Page properties: `page.*`

Example: `page.id` or `page.trail`.

> ### Current request properties: `request.*` 

Reviewer: Remove

Example: `request.query.get('foobar')`.

> ### Tag chunks: `tag.*`

Reviewer: Remove
